lib/config/app-insights-logger.js

```diff
@@ -0,0 +1,24 @@
+const applicationinsights = require('applicationinsights');
+
+if (process.env.APPINSIGHTS_INSTRUMENTATIONKEY) {
+  applicationinsights.setup(process.env.APPINSIGHTS_INSTRUMENTATIONKEY).setAutoCollectConsole(true);
+  applicationinsights.start();
+}
+
+module.exports.AppInsightsLogger = class AppInsightsLogger {
+  constructor() {
+    this.client = applicationinsights.defaultClient;
+  }
+  error(...msgs) {
+    this.client.trackException({ exception: new Error(msgs.filter(item => item).map(JSON.stringify).join(' ')) });
+  }
+  warn(...msgs) {
+    this.client.trackTrace({ message: msgs.filter(item => item).map(JSON.stringify).join(' '), severity: 2 });
```

Comment:
what this "severity" values 0, 1, 2,means ? I thing for error we should set to 0 and warn as 1, debug, info can be 2, 3

Reply:
The different levels of severity in traces are

Verbose = 0 (Used for info)
Information = 1 (Used for debug)
Warning = 2 (Used for warn)
Error = 3
Critical = 4

For errors, `trackException` is used so that errors get logged in the failures section in dashboard.

Reply:
Yes, I think trackException is fine as we want errors to get logged in the Failures section in dashboard.

@s-anupam please add comments for severity values

Reply:
Added comments